src/vs/platform/terminal/common/capabilities/commandDetection/promptInputModel.ts

```diff
@@ -555,7 +559,11 @@ export class PromptInputModel extends Disposable implements IPromptInputModel {
 				}
 			}
 			// Calculate the ghost text start index
-			ghostTextIndex = positionsWithGhostStyle[0] - this._commandStartX;
+			if (buffer.cursorY === this._commandStartMarker?.line) {
```

Comment:
Do you mean `(buffer.baseY + buffer.cursorY === this._commandStartMarker?.line)`? I expect this to fail if there is any scrollback otherwise